components/overflow-group/overflow-group.js

```diff
@@ -32,7 +32,7 @@ const OPENER_STYLE = {
 
 function createMenuItem(node) {
 	const childText = node.text || node.firstChild && (node.firstChild.label || node.firstChild.text || node.firstChild.textContent.trim());
-	const disabled = node.disabled || node.nonInteractive;
+	const disabled = node.disabled;
```

Comment:
I wasn't really sure what was going on here and why overflow-group cared about `nonInteractive` (or why the action had a non-interactive attribute). Maybe I'm missing something though.

Reply:
So yeah, maybe `nonInteractive` isn't needed on the action as-is.  The `nonInteractive` property wasn't meant to be set by a consumer, but rather internally depending on the selection state.  The `disabled` property is one that consumers usually expect to able to set, so I felt that wasn't quite right.  I was on the fence as to whether the property used should be public or private.  Maybe we can use `disabled` and provide a getter but no setter.

The `d2l-non-interactive-change` event was being used to toggle the `disabled` state of the corresponding menu-item inside the overflow menu.  This is something that `d2l-overflow-group` doesn't handle so I think we'll need to deal with that one way or another.  

The event is convenient but limited.  An alternative could be to wire-up `MutationObserver`s to the slotted buttons/actions.  However, `MutationObserver`s do not detect composed subtree changes - they only detect subtree changes within the context of the target.  I think to make that work, it would be necessary to wire-up the observers directly to the assigned (and flattened) nodes of the slot, watching for `disabled` attribute changes, and do this in conjunction with `slotchange`.  That might be a good sol'n that would also address this current gap with `d2l-overflow-group` and `disabled` state of buttons.

Reply:
Yeah I was thinking about this more over the weekend and landed on the `MutationObserver` approach as well. There are other things about the buttons that could change (like the text) that in theory overflow-group should be handling.